iperf-server-start

```diff
@@ -176,31 +178,55 @@ ss -tlnp
 echo
 #TODO: exit with error if ports still used
 
-ifname_numa_node=-1
-ifname_numa_node_cpus=-1
+ifname_numa_node=
+ifname_numa_node_cpus=
 if [ -e /sys/class/net/${ifname}/device/numa_node ]; then
     ifname_numa_node=$(cat /sys/class/net/${ifname}/device/numa_node)
-
-    ifname_numa_node_cpus=$(cat /sys/devices/system/node/node${ifname_numa_node}/cpulist)
-elif [ "${cpu_pin}" == "numa" ]; then
-    exit_error "cannot determine numa node for interface '${ifname}' and cpu-pin is numa"
+    if [ "$ifname_numa_node" == "-1" ]; then
+        # VM case, use numa 0
+        ifname_numa_node=0
+    fi
+else
+    # When topo=internode topo, the device is eth0. Default it to numa node0.
+    # In other word, if there is no numa associated with ifname, just use numa 0
+    ifname_numa_node=0
 fi
 
+ifname_numa_node_cpus=$(cat /sys/devices/system/node/node${ifname_numa_node}/cpulist)
+# convert to comma separated list if there are ranges i.e 0-5 items
+ifname_numa_node_cpus=$(echo $ifname_numa_node_cpus | perl -pe 's/(\d+)-(\d+)/join(",",$1..$2)/eg')
 echo "ifname=${ifname}"
 echo "ifname_numa_node=${ifname_numa_node}"
+echo "ifname_numa_node_cpus=${ifname_numa_node_cpus}"
+
+function assign-pin-cpu {
+    local my_cpu
+
+    # iperf PIN is to pin same NUMA and 1 CPU
+    IFS=',' read -r -a array <<< "$ifname_numa_node_cpus"
+    if [ "$1" == "client" ]; then
+        my_cpu=${array[${#array[@]}-1]}
+    else
+        my_cpu=${array[${#array[@]}-2]}
+    fi
+    cpu_pin_cmd="taskset --cpu-list ${my_cpu}"
+    echo $1: $cpu_pin_cmd
+}
```

Comment:
Since you have written this function twice now it seems like the kind of thing that should go in the `iperf-base` script.

Reply:
Good idea. Moved function to iperf-base.